src/test/deploy/functions/containerCleaner.spec.ts

```diff
@@ -209,3 +209,212 @@ describe("ContainerRegistryCleaner", () => {
     expect(stub.rm).to.not.have.been.called;
   });
 });
+
+describe("listGcfPaths", () => {
+  const LOCATIONS_US = Promise.resolve({
+    children: ["us-central1", "us-west2"],
+    digests: [],
+    tags: [],
+  });
+
+  const LOCATIONS_EU = Promise.resolve({
+    children: ["europe-west1", "europe-central2"],
+    digests: [],
+    tags: [],
+  });
+
+  const LOCATIONS_ASIA = Promise.resolve({
+    children: ["asia-northeast1", "asia-south1"],
+    digests: [],
+    tags: [],
+  });
+
+  const EMPTY = Promise.resolve({
+    children: [],
+    digests: [],
+    tags: [],
+  });
+
+  it("should throw an error on invalid location", () => {
+    expect(() => containerCleaner.listGcfPaths("project", ["invalid"])).to.throw;
+  });
+
+  it("should throw an error when subdomains fail search", () => {
+    const stub = sinon.createStubInstance(containerCleaner.DockerHelper);
+    const helpers = {
+      us: stub,
+      eu: stub,
+      asia: stub,
+    };
+
+    expect(async () => await containerCleaner.listGcfPaths("project", undefined, helpers)).to.throw;
+  });
+
+  it("should list paths, single location param", async () => {
+    const stub = sinon.createStubInstance(containerCleaner.DockerHelper);
+    stub.ls.withArgs("project/gcf").returns(LOCATIONS_US);
+    const stubEmpty = sinon.createStubInstance(containerCleaner.DockerHelper);
+    stubEmpty.ls.withArgs("project/gcf").returns(EMPTY);
+    const helpers = { us: stub, eu: stubEmpty, asia: stubEmpty };
+
+    const paths = await containerCleaner.listGcfPaths("project", ["us-central1"], helpers);
+
+    expect(paths.length).to.eq(1);
```

Comment:
FYI. You can use `deep` here (though you'd have to use `sort` if the list has multiple elements.

```
expect(paths).to.deep.equal(["us.gcr.io/projects/gcf/us-central1"]);
```

Reply:
Changed to use `deep` in all occurrences.